doc/content/source/filters/EnergyFilter.md

```diff
@@ -0,0 +1,27 @@
+# EnergyFilter
+
+!alert note
+`EnergyFilter` can only be added to problems when the input files contains a [OpenMCCellAverageProblem](OpenMCCellAverageProblem.md)
+in the `[Problem]` block. Otherwise, attempting to add this filter will result in an error.
+
+## Description
+
+The `EnergyFilter` object provides a thin wrapper around an OpenMC `EnergyFilter` which local tallies (added through the
+[tally system](AddTallyAction.md)) can access. This filter bins particles based off of the particle energy before the collision
+has occured (for collision or analog estimators) or the particle energy (for tracklength estimators). The ordering of the bins in
+`EnergyFilter` can be reversed (set to be descending in terms of energy) by setting `reverse_bins = true` if you wish to use
+the conventional multi-group radiation transport formalism.
+
+## Example Input File Syntax
+
+As an example, a [MeshTally](MeshTally.md) named `Flux` applies an `EnergyFilter` named `Energy` to break the total flux up
+into multi-group fluxes using the CASMO-2 group structure.
```

Comment:
Can we expose the built-in group structures in OpenMC? So that if the user wants many groups, they don't need to type them out, but could use the enumeration from OpenMC?

Reply:
@aprilnovak AFAIK the built-in group structures in OpenMC are purely on the [Python side of the API](https://github.com/openmc-dev/openmc/blob/develop/openmc/mgxs/__init__.py) so there's no "easy" way of exposing them through OpenMC headers. I could copy and paste the structures into C++ code (as a constexpr array) if you're OK with the `EnergyFilter.h` getting rather bloated, but I can't think of a more streamlined way to go about this...

I think it's still worthwhile to add, just wanted to make sure you're OK with this implementation approach before I start pasting the structures in.

Reply:
I think pasting them in is fine -- I assume they are not going to be revised in the future so it'll be a lot of line changes now, but shouldn't be something we need to really maintain a bunch.